encoder/vaapiencoder_vp8.cpp

```diff
@@ -49,14 +50,303 @@ class VaapiEncPictureVP8 : public VaapiEncPicture
     }
 };
 
-VaapiEncoderVP8::VaapiEncoderVP8():
-	m_frameCount(0),
-	m_qIndex(VP8_DEFAULT_QP)
+class IVaapiFlagParameter {
+public:
+    virtual bool fillPictureParameter(VAEncPictureParameterBufferVP8*, uint8_t) const = 0;
+    virtual bool fillLayerID(VAEncMiscParameterTemporalLayerStructure*) const = 0;
+    virtual bool fillLayerBitrate(VAEncMiscParameterRateControl*, uint32_t temporalId) const = 0;
+    virtual bool fillLayerFramerate(VAEncMiscParameterFrameRate* frameRate, uint32_t temporalId) const = 0;
+
+    virtual int8_t getErrorResilient() const = 0;
+    virtual int8_t getRefreshEntropyProbs() const = 0;
+    virtual uint8_t getTemporalLayer(uint32_t frameNum) const = 0;
+};
+
+class VaapiFlagParameterNormal : public IVaapiFlagParameter {
+public:
+    virtual bool fillPictureParameter(VAEncPictureParameterBufferVP8* pictureParameter, uint8_t temporalLayer) const;
+    virtual bool fillLayerID(VAEncMiscParameterTemporalLayerStructure* temporalLayer) const { return true; }
+    virtual bool fillLayerBitrate(VAEncMiscParameterRateControl* rateControl, uint32_t temporalId) const { return true; }
+    virtual bool fillLayerFramerate(VAEncMiscParameterFrameRate* frameRate, uint32_t temporalId) const { return true; }
+
+    virtual int8_t getErrorResilient() const { return 0; }
+    virtual int8_t getRefreshEntropyProbs() const { return 0; }
+    virtual uint8_t getTemporalLayer(uint32_t frameNum) const { return 0; }
+};
+
+class VaapiFlagParameterSVCT : public IVaapiFlagParameter {
+public:
+    VaapiFlagParameterSVCT(const SVCTVideoFrameRate& framerates, const uint32_t* layerBitrate);
+
+public:
+    virtual bool fillPictureParameter(VAEncPictureParameterBufferVP8* pictureParameter, uint8_t temporalLayer) const;
+    virtual bool fillLayerID(VAEncMiscParameterTemporalLayerStructure* temporalLayer) const;
+    virtual bool fillLayerBitrate(VAEncMiscParameterRateControl* rateControl, uint32_t temporalId) const;
+    virtual bool fillLayerFramerate(VAEncMiscParameterFrameRate* frameRate, uint32_t temporalId) const;
+
+    virtual int8_t getErrorResilient() const { return 1; }
+    virtual int8_t getRefreshEntropyProbs() const { return 0; }
+    virtual uint8_t getTemporalLayer(uint32_t frameNum) const { return m_tempLayerIDs[frameNum % m_periodicity]; }
+
+public:
+    void printRatio();
+    void printLayerIDs();
+
+protected:
+    bool calculateFramerateRatio();
+    bool calculatePeriodicity();
+    bool calculateLayerIDs();
+    uint32_t calculateGCD(uint32_t* gcdArray, uint32_t num);
+
+private:
+    VideoFrameRate m_framerates[VP8_MAX_TEMPORAL_LAYER_NUM];
+    uint32_t m_framerateRatio[VP8_MAX_TEMPORAL_LAYER_NUM];
+    uint32_t m_layerBitRate[VP8_MAX_TEMPORAL_LAYER_NUM];
+    uint32_t m_periodicity;
+    Uint32Vector m_tempLayerIDs;
+    uint8_t m_layerNum;
+};
+
```

Comment:
Hi @lizhong1008 ,
The class only used by this .cpp, it's no need add this to .h.